rs/tests/testnets/large_high_perf.rs

```diff
@@ -0,0 +1,172 @@
+//
+//     "prometheus": "Prometheus Web UI at http://prometheus.large_high_perf--1692597750709.testnet.farm.dfinity.systems",
+//     "grafana": "Grafana at http://grafana.large_high_perf--1692597750709.testnet.farm.dfinity.systems",
+//     "progress_clock": "IC Progress Clock at http://grafana.large_high_perf--1692597750709.testnet.farm.dfinity.systems/d/ic-progress-clock/ic-progress-clock?refresh=10s\u0026from=now-5m\u0026to=now",
+//
+// Happy testing!
+
+use anyhow::Result;
+
+use ic_consensus_system_test_utils::rw_message::install_nns_with_customizations_and_check_progress;
+use ic_registry_subnet_type::SubnetType;
+use ic_system_test_driver::driver::ic::{
+    AmountOfMemoryKiB, ImageSizeGiB, InternetComputer, NrOfVCPUs, Subnet, VmResources,
+};
+use ic_system_test_driver::driver::ic_gateway_vm::{HasIcGatewayVm, IcGatewayVm};
+use ic_system_test_driver::driver::pot_dsl::PotSetupFn;
+use ic_system_test_driver::driver::vector_vm::VectorVm;
+use ic_system_test_driver::driver::{
+    farm::HostFeature,
+    group::SystemTestGroup,
+    prometheus_vm::{HasPrometheus, PrometheusVm},
+    test_env::TestEnv,
+    test_env_api::HasTopologySnapshot,
+};
+use nns_dapp::{nns_dapp_customizations, set_authorized_subnets, set_icp_xdr_exchange_rate};
+use slog::info;
+
+const NUM_IC_GATEWAYS: u64 = 1;
+const DEFAULT_IMAGE_SIZE_GIB: u64 = 5120;
+
+fn main() -> Result<()> {
+    // No default value is set for PERF_HOSTS to ensure users consciously select dedicated performance hosts and understand their significance.
+    let perf_hosts = std::env::var("PERF_HOSTS").unwrap_or_else(|_| {
+        panic!("PERF_HOSTS environment variable must be set (comma-separated list of host names)")
+    });
+
+    // By default, the image size is set to 5 TiB, which supports testing up to 2 TiB of state under heavy write workloads.
+    // Note: Migrating such a large image to the LVM partition on the hosts can be time-consuming.
+    // To use a different image size, set the IMAGE_SIZE_GIB environment variable.
+    let image_size_gib = std::env::var("IMAGE_SIZE_GIB")
+        .ok()
+        .and_then(|s| s.parse::<u64>().ok())
+        .unwrap_or(DEFAULT_IMAGE_SIZE_GIB);
+
+    let perf_hosts: Vec<String> = perf_hosts.split(',').map(|s| s.to_string()).collect();
+    let config = Config::new(perf_hosts, image_size_gib);
+
+    SystemTestGroup::new()
+        .with_setup(config.build())
+        .execute_from_args()?;
+    Ok(())
+}
+
+#[derive(Clone, Debug)]
+pub struct Config {
+    hosts: Vec<String>,
+    image_size_gib: u64,
+}
+
+impl Config {
+    pub fn new(hosts: Vec<String>, image_size_gib: u64) -> Config {
+        Config {
+            hosts,
+            image_size_gib,
+        }
+    }
+
+    /// Builds the IC instance.
+    pub fn build(self) -> impl PotSetupFn {
+        move |env: TestEnv| setup(env, self)
+    }
+}
+
+pub fn setup(env: TestEnv, config: Config) {
+    // start p8s for metrics and dashboards
+    PrometheusVm::default()
```

Comment:
It would be good if the PrometheusVm and the VectorVm will be located in dm1 as well since that's where the PERF_HOSTS are located. So I would suggest configuring the required `HostFeature ::Performance` for these VMs.